### PR TITLE
Add Windows Gather Power Shell History module

### LINKED ISSUE #10979: Add Windows Gather User History module
Add post module to collect [PSReadline](https://blogs.technet.microsoft.com/heyscriptingguy/2014/06/18/better-powershell-history-management-with-psreadline/) history, plus anything else that may be of interest (`*_history.txt`), in a similar manner to [Linux Gather User History](https://github.com/rapid7/metasploit-framework/tree/master/modules/post/linux/gather/enum_users_history.rb).

`%USERPROFILE%\AppData\Roaming\Microsoft\Windows\PowerShell\PSReadline\ConsoleHost_history.txt`

* https://twitter.com/nikhil_mitt/status/1062382974744887296
* https://docs.microsoft.com/en-us/powershell/module/psreadline/
* https://github.com/KalibRx/PoshHarvestPy/blob/master/poshharvest.py
* https://0xdf.gitlab.io/2018/11/08/powershell-history-file.html

### PR DESCRIPTION
**Pull request by dgarvit**
Resolves #10979.

### CODE REVIEW

File: modules/post/windows/gather/psreadline_history.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        'Author' => [
+          'Garvit Dewan <d.garvit[at]gmail.com>' # @dgarvit
+        ],
+        'Platform' => %w{ win },
+        'SessionTypes' => [ 'meterpreter' ]
+      ))
+  end
+
+  def run
+
```
Comment (wvu-r7) [inline modules/post/windows/gather/psreadline_history.rb]: Generally, we PR something WIP when it works in at least one specific test case, but there's nothing here. Are there commits you haven't pushed? :)

Comment (dgarvit) [reply modules/post/windows/gather/psreadline_history.rb]: oh, i just meant work in progress. I'll remove the tag. thanks for pointing out!

Comment (wvu-r7) [reply modules/post/windows/gather/psreadline_history.rb]: No, it's still WIP. But you've PR'd empty code. :)

Comment (dgarvit) [reply modules/post/windows/gather/psreadline_history.rb]: Okay, I'll close this for now. I'll create a PR when it works in at least one specific test case. :)

Comment (wvu-r7) [reply modules/post/windows/gather/psreadline_history.rb]: Sounds good to me! We just want something that's testable. :)

**Comment by bcoles** [comment]
#11151

**Pull request closed without merging**